src/v/archival/ntp_archiver_service.cc

```diff
@@ -361,18 +363,30 @@ ss::future<ntp_archiver::batch_result> ntp_archiver::upload_next_candidates(
   std::optional<model::offset> lso_override) {
     retry_chain_logger ctxlog(archival_log, parent, _ntp.path());
     vlog(ctxlog.debug, "Uploading next candidates called for {}", _ntp);
+    if (_gate.is_closed()) {
+        return ss::make_ready_future<batch_result>(batch_result{});
```

Comment:
BTW fields in `batch_result` are not default-initialized

Reply:
It's zero initialised and the code was removed in upstream already

Reply:
cool, didn't know it!